tests/load/bigquery/test_bigquery_table_builder.py

```diff
@@ -371,3 +380,471 @@ def demo_source() -> DltResource:
+
+    hints = bigquery_adapter(no_hints._clone(new_name="hints"), cluster="col1")
+
+    @dlt.source(max_table_nesting=0)
+    def sources() -> List[DltResource]:
+        return [no_hints, hints]
+
+    pipeline = destination_config.setup_pipeline(
+        f"bigquery_{uniq_id()}",
+        full_refresh=True,
+    )
+
+    # noinspection PyArgumentList
+    pipeline.run(sources())
+
+    with pipeline.sql_client() as c:
+        nc: google.cloud.bigquery.client.Client = c.native_connection
+
+        fqtn_no_hints = c.make_qualified_table_name("no_hints", escape=False)
+        fqtn_hints = c.make_qualified_table_name("hints", escape=False)
+
+        no_hints_table = nc.get_table(fqtn_no_hints)
+        hints_table = nc.get_table(fqtn_hints)
+
+        no_hints_cluster_fields = (
+            [] if no_hints_table.clustering_fields is None else no_hints_table.clustering_fields
+        )
+        hints_cluster_fields = (
+            [] if hints_table.clustering_fields is None else hints_table.clustering_fields
+        )
+
+        assert not no_hints_cluster_fields, "`no_hints` table IS clustered by `col1`."
+        assert ["col1"] == hints_cluster_fields, "`hints` table IS NOT clustered by `col1`."
+
+
+def test_adapter_hints_empty() -> None:
+    @dlt.resource(columns=[{"name": "int_col", "data_type": "bigint"}])
+    def some_data() -> Iterator[Dict[str, str]]:
+        yield from next(sequence_generator())
+
+    with pytest.raises(
+        ValueError,
+        match="^AT LEAST one of `partition`, `cluster`, `round_half_away_from_zero`",
+    ):
+        bigquery_adapter(some_data)
+
+
+def test_adapter_hints_round_mutual_exclusivity_requirement() -> None:
+    @dlt.resource(columns=[{"name": "double_col", "data_type": "double"}])
+    def some_data() -> Iterator[Dict[str, str]]:
+        yield from next(sequence_generator())
+
+    with pytest.raises(
+        ValueError,
+        match=(
+            "are present in both `round_half_away_from_zero` and `round_half_even` "
+            "which is not allowed. They must be mutually exclusive.$"
+        ),
+    ):
+        bigquery_adapter(
+            some_data, round_half_away_from_zero="double_col", round_half_even="double_col"
+        )
+
+
+def test_adapter_hints_parsing_table_description() -> None:
+    @dlt.resource(columns=[{"name": "double_col", "data_type": "double"}])
+    def some_data() -> Iterator[Dict[str, str]]:
+        yield from next(sequence_generator())
+
+    table_description = "Once upon a time a small table got hinted."
+    bigquery_adapter(some_data, table_description=table_description)
+
+    assert some_data._hints["x-bigquery-table-description"] == table_description  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(all_staging_configs=True, subset=["bigquery"]),
+    ids=lambda x: x.name,
+)
+def test_adapter_hints_table_description(destination_config: DestinationTestConfiguration) -> None:
```

Comment:
Is this a valuable hint?

If I should implement it, where can I find a handle to the DDL.

Reply:
I do not understand your question...